samples/v2/component_with_optional_inputs.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+from kfp import compiler
+from kfp import dsl
+from kfp.dsl import component
+
+
+@component
+def component_op(
+    input1: str = 'default value',
+    input2: Optional[str] = None,
```

Comment:
Do we support other types? If yes,m should we add them: `int`, `float`, `list`, `dict`, `set`, etc.

Reply:
Thank you, I added more types.